src/components/PeoplePage/PeoplePage.tsx

```diff
@@ -0,0 +1,88 @@
+import { useCallback, useEffect, useState } from 'react';
+import { Loader } from '../Loader';
+import { getPeople } from '../../api';
+import { Person } from '../../types';
+import { PeopleTable } from '../PeopleTable';
+
+const findPersonByName = (
+  peopleToFindFrom: Person[],
+  nameToFind: string | null,
+) => {
+  if (nameToFind) {
+    return peopleToFindFrom.find(({ name }) => name === nameToFind);
+  }
+
+  return undefined;
+};
+
+export const PeoplePage: React.FC = () => {
+  const [people, setPeople] = useState<Person[]>([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [isError, setIsError] = useState(false);
+  const [isListEmpty, setIsListEmpty] = useState(false);
+
+  const loadPeople = useCallback(async () => {
+    setIsLoading(true);
+    try {
+      const peopleFromServer = await getPeople();
+
+      const preparedPeople = peopleFromServer.map(person => {
+        const personCopy = { ...person };
+
+        personCopy.mother = findPersonByName(
+          peopleFromServer,
+          person.motherName,
+        );
+        personCopy.father = findPersonByName(
+          peopleFromServer,
+          person.fatherName,
+        );
+
+        return personCopy;
+      });
+
+      setPeople(preparedPeople);
+
+      if (!peopleFromServer.length) {
```

Comment:
why do you need it?
You can handle it without extra state

Reply:
I wanted not to show this message when people are loading from the server for the first time, because before that it is always an empty array and this message will be shown when people are loading

Reply:
But you have put a two check `...length && !loading`